src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -118,6 +294,140 @@
+     *     <li>Otherwise, return {@code otherStatus}.</li>
+     * </ul>
+     * <p>
+     * Example, to create a filter that will reject any class loaded from the application classloader.
+     * <pre><code>
+     *     ObjectInputFilter f = rejectFilter(cl ->
+     *          cl.getClassLoader() == ClassLoader.ClassLoader.getSystemClassLoader(), Status.UNDECIDED);
+     * </code></pre>
+     *
+     * @param predicate a predicate to test a non-null Class, non-null
+     * @param otherStatus a Status to use if the predicate is {@code false}
+     * @return returns a filter that returns {@code REJECTED} if the predicate
+     *          on the class is {@code true}
+     * @since 17
+     */
+    static ObjectInputFilter rejectFilter(Predicate<Class<?>> predicate, Status otherStatus) {
+        Objects.requireNonNull(predicate, "predicate");
+        Objects.requireNonNull(otherStatus, "otherStatus");
+        return new Config.PredicateFilter(predicate, REJECTED, otherStatus);
+    }
+
+    /**
+     * Returns a filter that merges the status of a filter and another filter.
+     * If {@code another} filter is {@code null}, the {@code filter} is returned.
+     * Otherwise, a {@code filter} is returned to merge the pair of {@code non-null} filters.
+     *
+     * The filter returned implements the {@link ObjectInputFilter#checkInput(FilterInfo)} method
+     * as follows:
+     * <ul>
+     *     <li>Invoke {@code filter} on the {@code FilterInfo} to get its {@code status};
+     *     <li>Return {@code REJECTED} if the {@code status} is {@code REJECTED};
+     *     <li>Invoke {@code anotherFilter} to get the {@code otherStatus};
+     *     <li>Return {@code REJECTED} if the {@code otherStatus} is {@code REJECTED};
+     *     <li>Return {@code ALLOWED}, if either {@code status} or {@code otherStatus}
+     *          is {@code ALLOWED}, </li>
+     *     <li>Otherwise, return {@code UNDECIDED}</li>
+     * </ul>
+     *
+     * @param filter a filter, non-null
+     * @param anotherFilter a filter to be merged with the filter, may be {@code null}
+     * @return an {@link ObjectInputFilter} that merges the status of the filter and another filter
+     * @since 17
+     */
+    static ObjectInputFilter merge(ObjectInputFilter filter, ObjectInputFilter anotherFilter) {
+        Objects.requireNonNull(filter, "filter");
+        return (anotherFilter == null) ? filter : new Config.MergeFilter(filter, anotherFilter);
+    }
+
+    /**
+     * Returns a filter that invokes a given filter and maps {@code UNDECIDED} to {@code REJECTED}
+     * for classes, with some special cases, and otherwise returns the status.
+     * The filter returned checks that classes not {@code ALLOWED} by the filter
+     * are {@code REJECTED}.  Either the class is not {@code ALLOWED} or
+     * if the class is an array and the base component type is not allowed,
+     * otherwise the result is {@code UNDECIDED}.
```

Comment:
There is an extra "if" at line 395.
it should be a more readable version of the list below implementing checkfilter.
If it does not aid in understanding, it can be removed.